src/lib/teacher-network.ts

```diff
@@ -67,31 +67,52 @@ export function syncTeacherClassesAndOfferings(firestore: Firestore, user: UserM
     promises.push(syncClass(firestore, rawPortalJWT, userClasses[context_id]));
   });
 
-  // synchronize the offerings
-  user.portalClassOfferings.forEach(async offering => {
-    const {
-      offeringId: id, activityTitle: name, activityUrl: uri, classHash: context_id, classUrl,
-      unitCode: unit, problemOrdinal: problem
-    } = offering;
-    const problemPath = getProblemPath(unit, problem);
-    const fsOffering: OfferingWithoutTeachers = { id, name, uri, context_id, unit, problem, problemPath, network };
-    promises.push(syncOffering(firestore, rawPortalJWT, classUrl, fsOffering));
-  });
+  if (network) {
+    // synchronize the offerings
+    user.portalClassOfferings.forEach(async offering => {
+      const {
+        offeringId: id, activityTitle: name, activityUrl: uri, classHash: context_id, classUrl,
+        unitCode: unit, problemOrdinal: problem
+      } = offering;
+      const problemPath = getProblemPath(unit, problem);
+      const fsOffering: OfferingWithoutTeachers = { id, name, uri, context_id, unit, problem, problemPath, network };
+      promises.push(syncOffering(firestore, rawPortalJWT, classUrl, fsOffering));
+    });
+  }
+  return Promise.all(promises);
+}
 
-  return promises;
+async function createOrUpdateClassDoc(firestore: Firestore, docPath: string, aClass: ClassDocument):
+      Promise<void|ClassDocument> {
+  return firestore.guaranteeDocument(docPath,
+     async () => { return aClass; },
+     (content) => { return !content || !arraysEqualIgnoringOrder(aClass.teachers, content.teachers); }
+  );
```

Comment:
If we're going to use `guaranteeDocument` here, we should update it to wrap the `get` and `set` in a transaction. Otherwise it'd be possible for something else to write out a class doc after the `get` is called but before the `set` is called. So then this previously written doc will get replaced by the one set here.
If we do change it to use a transaction we'll have to be careful to make sure the `writeContent` function has no important side effects since it might run multiple times.

It looks like `guaranteeDocument` will preserve the list of `networks` in the class document because it uses: `docRef.set(_content, { merge: true })`. So that part is good.